File: android/java/org/chromium/chrome/browser/brave_leo/BraveLeoSettingsLauncherHelper.java
/* Copyright (c) 2023 The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

package org.chromium.chrome.browser.brave_leo;

import android.app.Activity;
import android.content.Context;

import org.jni_zero.CalledByNative;

import org.chromium.chrome.browser.profiles.Profile;
import org.chromium.chrome.browser.settings.BraveLeoPreferences;
import org.chromium.chrome.browser.settings.SettingsNavigationFactory;
import org.chromium.chrome.browser.util.TabUtils;
import org.chromium.components.browser_ui.settings.SettingsNavigation;
import org.chromium.content_public.browser.WebContents;
import org.chromium.ui.base.WindowAndroid;

/** Launches Brave Leo settings page or subscription. */
public class BraveLeoSettingsLauncherHelper {
    private static final String ACCOUNT_PAGE_URL = "https://account.brave.com/";
    private static SettingsNavigation sLauncher;

    @CalledByNative
    private static void showBraveLeoSettings(WebContents webContents) {
        Context context = webContents.getTopLevelNativeWindow().getActivity().get();
        if (context == null) {
            return;
        }
        getLauncher().startSettings(context, BraveLeoPreferences.class);
    }

    @CalledByNative
    private static void goPremium(WebContents webContents) {
        BraveLeoUtils.goPremium(webContents.getTopLevelNativeWindow().getActivity().get());
    }

    @CalledByNative
    private static void managePremium(WebContents webContents) {
        if (BraveLeoPrefUtils.getIsSubscriptionActive(Profile.fromWebContents(webContents))) {
            BraveLeoUtils.openManageSubscription();
        } else {
            TabUtils.openURLWithBraveActivity(ACCOUNT_PAGE_URL);
        }
    }

    @CalledByNative
    private static void openURL(String url) {
        TabUtils.openURLWithBraveActivity(url);
    }

    @CalledByNative
    private static void handleVoiceRecognition(WebContents webContents, String conversation_uuid) {
        new BraveLeoVoiceRecognitionHandler(
                        webContents.getTopLevelNativeWindow(), webContents, conversation_uuid)
                .startVoiceRecognition();
    }

    @CalledByNative
    private static void closeActivity(WebContents webContents) {
        WindowAndroid windowAndroid = webContents.getTopLevelNativeWindow();
        if (windowAndroid == null) {
            return;
        }
        Activity activity = windowAndroid.getActivity().get();
        if (activity == null) {
            return;
        }
        activity.finish();
    }

    private static SettingsNavigation getLauncher() {
        return sLauncher != null ? sLauncher : SettingsNavigationFactory.createSettingsNavigation();
    }
}

File: browser/ui/android/ai_chat/brave_leo_settings_launcher_helper.cc
// Copyright (c) 2023 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

#include "brave/browser/ui/android/ai_chat/brave_leo_settings_launcher_helper.h"

#include "base/android/jni_android.h"
#include "base/android/jni_string.h"
#include "brave/build/android/jni_headers/BraveLeoSettingsLauncherHelper_jni.h"
#include "content/public/browser/web_contents.h"

namespace ai_chat {

void ShowBraveLeoSettings(content::WebContents* web_contents) {
  Java_BraveLeoSettingsLauncherHelper_showBraveLeoSettings(
      base::android::AttachCurrentThread(), web_contents->GetJavaWebContents());
}

void GoPremium(content::WebContents* web_contents) {
  Java_BraveLeoSettingsLauncherHelper_goPremium(
      base::android::AttachCurrentThread(), web_contents->GetJavaWebContents());
}

void ManagePremium(content::WebContents* web_contents) {
  Java_BraveLeoSettingsLauncherHelper_managePremium(
      base::android::AttachCurrentThread(), web_contents->GetJavaWebContents());
}

void OpenURL(const std::string& url) {
  Java_BraveLeoSettingsLauncherHelper_openURL(
      base::android::AttachCurrentThread(),
      base::android::ConvertUTF8ToJavaString(
          base::android::AttachCurrentThread(), url));
}

void HandleVoiceRecognition(content::WebContents* web_contents,
                            const std::string& conversation_uuid) {
  Java_BraveLeoSettingsLauncherHelper_handleVoiceRecognition(
      base::android::AttachCurrentThread(), web_contents->GetJavaWebContents(),
      base::android::ConvertUTF8ToJavaString(
          base::android::AttachCurrentThread(), conversation_uuid));
}

void CloseActivity(content::WebContents* web_contents) {
  Java_BraveLeoSettingsLauncherHelper_closeActivity(
      base::android::AttachCurrentThread(), web_contents->GetJavaWebContents());
}

}  // namespace ai_chat
